modules/cats-effect/src/test/scala/pureconfig/module/catseffect/CatsEffectSuite.scala

```diff
@@ -14,6 +14,11 @@ import pureconfig.{BaseSuite, ConfigReader, ConfigSource, ConfigWriter}
 final class CatsEffectSuite extends BaseSuite {
 
   case class SomeCaseClass(somefield: Int, anotherfield: String)
+
+  object SomeCaseClass {
+    def unapply(c: SomeCaseClass): Option[(Int, String)] = Some((c.somefield, c.anotherfield))
+  }
```

Comment:
Instead of creating an `unapply` implementation that attempts to mimic Scala 2's behavior in Scala 3, I think it's preferable to define the `ConfigWriter` for `SomeCaseClass` in a way that is compatible with both Scala 2 and 3. Something like the following should do it:

```scala
implicit val writer: ConfigWriter[SomeCaseClass] =
  ConfigWriter.forProduct2("somefield", "anotherfield") { case SomeCaseClass(somefield, anotherfield) =>
    (somefield, anotherfield)
  }
```

Reply:
Fixed